Docker build fails with build-context set

**Describe the bug**
The docker build command in the Make file fails. It gets fixed if the line `#syntax=docker/dockerfile:1.4` is moved to the top

**To Reproduce**
- Create an empty directory
- Copy the Makefile and Dockerfile from the `templates' directory to the newly created directory
- Copy the `module1`, `module2` and `packages` directory
- Run `make init`
- Run `make all`

This can also be reproduced by running the following docker command instead of `make all`:
```
docker build . -f Dockerfile --target lint --build-context component=module1
```

**Expected behavior**
The docker build succeeds

**Screenshots**
```
$ make all
make COMPONENT="module1" lint
make[1]: Entering directory '/home/sathyanarays/hello-world-plugin'
cp .golangci.yaml module1
docker build . -f Dockerfile --target lint --build-context component=module1
[+] Building 0.1s (2/2) FINISHED                                                                                                                                                                                                                                                
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                       0.0s
 => => transferring dockerfile: 1.54kB                                                                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                            0.0s
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: unsupported frontend capability moby.buildkit.frontend.contexts
make[1]: *** [Makefile:49: lint] Error 1
make[1]: Leaving directory '/home/sathyanarays/hello-world-plugin'
make: *** [Makefile:24: "module1"?"module1"?"module1"] Error 2
```

**Desktop (please complete the following information):**
 ```
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Docker version**
```
$ docker version
Client: Docker Engine - Community
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:02:57 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:03 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
**Buildx version**
```
$ docker buildx version
github.com/docker/buildx v0.8.2-docker 6224def4dd2c3d347eee19db595348c50d7cb491
```

--- a/templates/Dockerfile
+++ b/templates/Dockerfile
@@ -1,7 +1,7 @@
+#syntax=docker/dockerfile:1.4
+
 # Copyright 2021 VMware, Inc. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
-
-#syntax=docker/dockerfile:1.4
 
 ARG BUILDER_BASE_IMAGE=golang:1.17
 

